syntax = "proto3";

package dataset;

message Data {
    float score = 1;
    int32 index = 2;
}

message DataSetInstance {
    bytes configuration_hash = 1;
    bytes service_hash = 2;
    map<string, Data> data = 3;  // string is a cnf.
}

message DataSet {
    // identifica el Solver con una configuracion determinada.( hash.value.hex() ) de sha3-256
    repeated DataSetInstance data = 1;
}